### DISCUSSION TITLE
[Python] PNG Image becomes corrupted after uploading to storage bucket

### DISCUSSION THREAD

**Discussion by luisedelgado** [supabase/supabase#26257] [Questions]
Hey all!

I have the following snippet for trying to upload a PNG image to a storage bucket using the Python client. The image shows up in the bucket dashboard but it's corrupted, and therefore you can't do anything with it. I've looked at this a good amount of time, and searched for online examples but I haven't been able to pinpoint why this isn't working. Any thoughts? Thanks in advance!

```
    file = base64.b64encode(open("test2.png", 'rb').read()).decode()
    bucket = supabase.storage.from_("test-images")
    bucket.upload(file=file,path="testfolder/test2.png", file_options={"content-type": "image/png"})
```

**Comment by silentworks** [comment]
You shouldn't do any base64 encoding on the file, just read it and save the file read value to the bucket. Here is some code sample from one of my projects I modified to match your example:

```python
with open("./test2.png", "+rb") as f:
            supabase.storage.from_("test-images").upload(
                path="testfolder/test2.png",
                file=f.read(),
                file_options={"content-type": "image/png"},
            )
```
